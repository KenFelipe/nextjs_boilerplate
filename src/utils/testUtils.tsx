import { render, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import { theme } from '../styles/theme'

type props = {
  children: React.ReactNode
}

const AllTheProviders: React.FC = ({ children }: props) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'

export { customRender as render }
