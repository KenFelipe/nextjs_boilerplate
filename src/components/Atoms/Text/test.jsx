import { render, screen } from '../../../utils/testUtils'
import { Text } from './'

it('render label correctly', () => {
  const { container } = render(<Text label="react-testing-library" />)

  expect(screen.getByText(/react-testing-library/)).toBeInTheDocument()
  expect(container).toMatchSnapshot()
})
