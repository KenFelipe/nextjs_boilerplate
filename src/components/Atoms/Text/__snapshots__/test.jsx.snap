// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render label correctly 1`] = `
.c0 {
  font-size: 3.2rem;
  color: pink;
}

<div>
  <div
    class="c0"
  >
    <p>
      react-testing-library
    </p>
  </div>
</div>
`;
